File: Data Structures & Algorithms/merge-intervals/submission-1.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int n = intervals.length;
        List<int[]> ans = new ArrayList<>();

        Arrays.sort(intervals,(a,b) -> {
            if(a[0]==b[0])return Integer.compare(a[1],b[1]);
            return Integer.compare(a[0],b[0]);
        });
        
        int start = intervals[0][0];
        int end = intervals[0][1];
        for(int i=1;i<n;i++){
            if(end<intervals[i][0]){
                ans.add(new int[]{start,end});
                start = intervals[i][0];
            }
            end = Math.max(intervals[i][1],end);
        }
        ans.add(new int[]{start,end});
        return ans.toArray(new int[ans.size()][]);
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:

    int replaceAndFindMax(const string &s, int k, int replace){
        int ans = 0;
        int start = 0;
        int end = 0;
        while(end<s.length()){
            if(s[end]==replace)end++;
            else{
                if(k>0){
                    k--;
                    end++;
                }
                else{
                    if(s[start]!=replace)k++;
                    start++;
                }
            }
            ans = max(ans,end-start);
        }
        return ans;
    }

    int characterReplacement(string s, int k) {
        int ans = 0;
        for(int i=0;i<26;i++){
            ans = max(ans,replaceAndFindMax(s,k,'A'+i));
        }
        return ans;
    }
};
